### DISCUSSION TITLE
Feature request: Allow registration and payment by two different people

### DISCUSSION THREAD

**Discussion by daisylb** [pretix/pretix#3936] [Ideas]
One of the problems we encountered with Pretix for PyCon AU 2023 was with attendees whose tickets had been purchased for them by someone else, usually by employers who purchased a block of several tickets for several attendees all at once. Some of these attendees had not seen the very clear, mandatory checkbox of "we require masks at this event" (because they hadn't gone through the registration flow), whereas others did not have a ticket or access to any of the emails we had sent up and just showed up empty handed, and various other permutations which all caused headaches at check-in time.

While it wouldn't solve 100% of the issues we faced, it would be good if Pretix had some officially supported way of allowing one person to register and a different person to pay. Some ways I've seen this solved by other platforms include:

- Linux.conf.au's bespoke ticketing platform used to ask you at the start of the process whether you were buying tickets for yourself or someone else.
    - If you picked "yourself", it would take you through the registration process, then ask you for payment. But one of the payment options was to generate a tokenised link, which you could then send to someone else (e.g. your accounts payable department or whoever), and they would be able to pay on your behalf using that link.
    - If you picked "someone else", the flow was reversed; you would pick the products and pay, and you'd get a tokenised link to send to the attendee to complete registration.
- Tito essentially does something similar but reversed; it requests payment upfront, and then most of the registration flow happens **after payment**. In that flow, you can allocate tickets you've purchased to other people, who get an email sent from Tito with a tokenised link they can use to complete registration.

Something we looked into when configuring Pretix last year was using the gift card functionality, but we found we couldn't restrict gift cards only to certain products. That meant that if e.g. a company generated five gift cards for the value of a Professional ticket, and distributed those gift cards to their staff, there was nothing stopping the gift card recipients from e.g. buying an Enthusiast ticket and six T-shirts.

**Comment by raphaelm** [comment]
This is something we thought about ~2 years ago, but it has since been requested very rarely, so we did not keep it on our priority list, so thanks for bringing it up again!

We'd be more likely to go down the Tito-like route, where you can "postpone personalization" of the tickets until after the purchase. There are however many hard design decisions related to this and a lot of added complexity since we'll need to see how we deal with "non-personalized tickets" ~everywhere.

Since it's large and it's hard and there are so many things on our roadmap, it's hard to promise anything here, but I would also very much like to see some functionality like this.
